lib/internal/blocklist.js

```diff
@@ -139,6 +143,55 @@ class BlockList {
     return Boolean(this[kHandle].check(address[kSocketAddressHandle]));
   }
 
+  /*
+  * @param {string[]} data
+  * @example
+  * const data = [
+  *   "Subnet: IPv4 192.168.1.0/24",
+  *   "Address: IPv4 10.0.0.5",
+  *   "Range: IPv4 192.168.2.1-192.168.2.10",
+  *   "Range: IPv4 10.0.0.1-10.0.0.10"
+  * ];
+  */
+  // TODO: Add IPv6 support
+  #parseIPInfo(data) {
+    for (const item of data) {
+      const subnetMatch = item.match(
+        /Subnet: IPv4 (\d{1,3}(?:\.\d{1,3}){3})\/(\d{1,2})/,
+      );
+      if (subnetMatch) {
+        const { 1: network, 2: prefix } = subnetMatch;
+        this.addSubnet(network, NumberParseInt(prefix));
+        continue;
+      }
+
+      const addressMatch = item.match(/Address: IPv4 (\d{1,3}(?:\.\d{1,3}){3})/);
+      if (addressMatch) {
+        const { 1: address } = addressMatch;
+        this.addAddress(address);
+        continue;
+      }
+
+      const rangeMatch = item.match(
+        /Range: IPv4 (\d{1,3}(?:\.\d{1,3}){3})-(\d{1,3}(?:\.\d{1,3}){3})/,
+      );
+      if (rangeMatch) {
+        const { 1: start, 2: end } = rangeMatch;
+        this.addRange(start, end);
+        continue;
+      }
+    }
+  }
+  toJson() {
+    return JSONStringify(this.rules, null, 2);
+  }
+  fromJson(data) {
+    if (typeof data === 'string') {
+      this.#parseIPInfo(JSONParse(data));
+    } else {
+      this.#parseIPInfo(data);
+    }
+  }
```

Comment:
```suggestion
  }

```

Just some style nits... please add some white space between the method bodies